src/library/scala/collection/immutable/Vector.scala

```diff
@@ -286,61 +271,46 @@ override def companion: GenericCompanion[Vector] = Vector
         s
       } else {
 
-        val freeSpace = ((1<<5*(depth)) - endIndex) // free space at the right given the current tree-structure depth
-        val shift = freeSpace & ~((1<<5*(depth-1))-1) // number of elements by which we'll shift right (only move at top level)
-        val shiftBlocks = freeSpace >>> 5*(depth-1) // number of top-level blocks
+        val freeSpace = ((1 << (5 * depth)) - endIndex) // free space at the right given the current tree-structure depth
+        val shift = (freeSpace & ~((1 << (5 * (depth - 1))) - 1)) // number of elements by which we'll shift right (only move at top level)
+        val shiftBlocks = (freeSpace >>> (5 * (depth - 1))) // number of top-level blocks
```

Comment:
Spaces are good.  Gratuitious parens around the entire expression are not good--just visual clutter that doesn't aid anything.  Internal parens that make one not rely upon knowing operator precedence are good.